command_prompt.bash

```diff
@@ -3,237 +3,153 @@
 }
 
-_command_prompt() {
-    # Run a pre-command if set.
-    if [[ -n $SEAFLY_PRE_COMMAND ]]; then
-        eval $SEAFLY_PRE_COMMAND
-    fi
+_seafly_prompt_command() {
+    eval $SEAFLY_PRE_COMMAND
 
-    # Set a simple prompt for non-interactive or non-color terminals.
-    if [[ $_interactive_terminal = 0 || $_color_terminal = 0 ]]; then
-        PS1='\h \w > '
-        return
-    fi
-
-    local prompt_prefix
-    if [[ -n $SEAFLY_PROMPT_PREFIX ]]; then
-        local prefix_value=$(eval $SEAFLY_PROMPT_PREFIX)
-        if [[ -n $prefix_value ]]; then
-            prompt_prefix="\[$SEAFLY_PREFIX_COLOR\]$prefix_value "
-        fi
-    fi
+    PS1=$(eval $SEAFLY_PROMPT_PREFIX)
+    PS1=${PS1:+"\[$SEAFLY_PREFIX_COLOR\]$PS1 "}
 
-    local prompt_start
-    if [[ $SEAFLY_SHOW_USER = 1 ]]; then
-        prompt_start="\[$SEAFLY_HOST_COLOR\]\u@\h"
-    else
-        prompt_start="\[$SEAFLY_HOST_COLOR\]\h"
-    fi
+    PS1+="\[$SEAFLY_HOST_COLOR\]"
+    (( SEAFLY_SHOW_USER )) && PS1+="\u@"
+    PS1+="\h"
 
     # Collate Git details, if applicable, for the current directory.
-    if [[ -z $GITSTATUS_DIR ]]; then
-        _git_details_fallback
-    else
-        _git_details_optimized
-    fi
+    _seafly_fetch_git_status && _seafly_format_git_status
 
-    local prompt_middle
-    if [[ $SEAFLY_LAYOUT = 1 ]]; then
-        prompt_middle="\[$SEAFLY_GIT_COLOR\]$SEAFLY_GIT_DETAILS\[$SEAFLY_PATH_COLOR\] \w"
+    if (( SEAFLY_LAYOUT )); then
+        PS1+="\[$SEAFLY_GIT_COLOR\]$_seafly_git_details\[$SEAFLY_PATH_COLOR\] \w"
     else
-        prompt_middle="\[$SEAFLY_PATH_COLOR\] \w\[$SEAFLY_GIT_COLOR\]$SEAFLY_GIT_DETAILS"
+        PS1+="\[$SEAFLY_PATH_COLOR\] \w\[$SEAFLY_GIT_COLOR\]$_seafly_git_details"
     fi
-    unset SEAFLY_GIT_DETAILS
+
+    unset _seafly_git_details
 
     # Normal prompt indicates that the last command ran successfully.
     # Alert prompt indicates that the last command failed.
-    local prompt_end="\$(if [[ \$? = 0 ]]; then echo \[\$SEAFLY_NORMAL_COLOR\]; else echo \[\$SEAFLY_ALERT_COLOR\]; fi) $SEAFLY_PROMPT_SYMBOL\[\$NOCOLOR\] "
+    _seafly_color=("$SEAFLY_ALERT_COLOR" "$SEAFLY_NORMAL_COLOR")
+    PS1+="\[\${_seafly_color[\$((!\$?))]}\] $SEAFLY_PROMPT_SYMBOL\[\$NOCOLOR\] "
 
-    PS1="$prompt_prefix$prompt_start$prompt_middle$prompt_end"
     PS2="\[$SEAFLY_NORMAL_COLOR\]$SEAFLY_PS2_PROMPT_SYMBOL\[\$NOCOLOR\] "
 }
 
 # Use [gitstatus](https://github.com/romkatv/gitstatus) if it is available.
-if [[ -d ~/.gitstatus ]]; then
-    export GITSTATUS_DIR=~/.gitstatus
-fi
-if [[ -f $GITSTATUS_DIR/gitstatus.prompt.sh ]]; then
-    source ~/.gitstatus/gitstatus.prompt.sh
-    gitstatus_stop && gitstatus_start
+if [[ -r "$SEAFLY_GITSTATUS_DIR"/gitstatus.plugin.sh ]]; then
+    source "$SEAFLY_GITSTATUS_DIR"/gitstatus.plugin.sh
+    gitstatus_stop && gitstatus_start -c 0 -d 0
```

Comment:
The flags tell gitstatus that we don't care about  conflicted and untracked files.